### PR TITLE
Add java 17 tools

### PR DESCRIPTION
**Pull request by timja**

### CODE REVIEW

**Review by olblak — DISMISSED** [review]

**Comment by timja** [comment]
Somethings not quite right

File: dist/profile/templates/azure.ci.jenkins.io/tools.yaml.erb [hunk 1]
```diff
@@ -56,6 +56,30 @@ tool:
...
               subdir: "jdk-<%= @tools["jdk11"]["version"] %>"
               url: "<%= @tools["jdk11"]["sourceURL"] %>/jdk-<%= @tools["jdk11"]["version"] %>/OpenJDK11U-jdk_s390x_linux_hotspot_<%= @tools["jdk11"]["version"].gsub('+', '_') %>.tar.gz"
+    - name: "jdk17"
+      properties:
+      - installSource:
+          installers:
+          - zip:
+              label: "linux && amd64"
+              subdir: "jdk-<%= @tools["jdk17"]["version"] %>"
+              url: "<%= @tools["jdk17"]["sourceURL"] %>/jdk-<%= @tools["jdk17"]["version"] %>/OpenJDK17-jdk_x64_linux_hotspot_<%= @tools["jdk17"]["version"].gsub('+', '_') %>.tar.gz"
```
Comment (dduportal) [inline dist/profile/templates/azure.ci.jenkins.io/tools.yaml.erb:66]: I got the resulting URL on vagrant, which is 404 Not found: https://github.com/adoptium/temurin11-binaries/releases/download/jdk-17+35/OpenJDK17-jdk_x64_linux_hotspot_17_35.tar.gz

**Review by dduportal — CHANGES_REQUESTED** [review]
Good job! Currently checking but it seems the URL is not completely OK

File: dist/profile/templates/azure.ci.jenkins.io/tools.yaml.erb (hunk 1 above)
Comment (timja) [reply dist/profile/templates/azure.ci.jenkins.io/tools.yaml.erb:66]: hmm s/11/17 somewhere

File: hieradata/vagrant/common.yaml [hunk 2]
```diff
@@ -164,6 +164,9 @@ profile::buildmaster::tools:
   jdk11:
     version: "11.0.12+7"
     sourceURL: https://github.com/adoptium/temurin11-binaries/releases/download
+  jdk17:
+    version: "17+35"
+    sourceURL: https://github.com/adoptium/temurin11-binaries/releases/download
```
Comment (timja) [inline hieradata/vagrant/common.yaml]:
```suggestion
    sourceURL: https://github.com/adoptium/temurin17-binaries/releases/download
```

File: dist/profile/templates/azure.ci.jenkins.io/tools.yaml.erb (hunk 1 above)
Comment (timja) [reply dist/profile/templates/azure.ci.jenkins.io/tools.yaml.erb:66]: fixed

**Review by dduportal — APPROVED** [review]
LGTM 🚀

**Pull request merged**